src/observer.ts

```diff
@@ -181,10 +197,38 @@ export class DataLayerObserver {
       return;
     }
 
+    const monitors: Monitor[] = [];
+
     try {
       const handler = this.addHandler(source);
       handler.debug = !!debug;
 
+      try {
+        if (monitor) {
+          // use select to get the target with the desired properties
+          // NOTE using [()] to pick, etc returns a copy and not the actual data layer reference
+          const target = select(source);
+
+          if (target) {
+            const braketPos = source.lastIndexOf('[(');
```

Comment:
I get nervous when I see string parsing of selectors happening outside of selector.ts. Perhaps a better way would be to export the `parsePath` method from that file? It returns a (memoized) Path instance and we could add the string parsing and formatting logic to that class.

Reply:
@TrevorFSmith I exported the `parsePath` and `ElementKind` so I could rely on it to walk the selector path better.  The one aspect that is not internal to `selector.ts` is the `Brackets` expose the `Op` `raw` value, which is something like `(cartID,price)` but not the surrounding bracket chars.  I could enhance `Bracket` to have this additional "raw brackets" value or we can keep `ref = select(source.substring(0, source.indexOf(`[${brackets.op.raw}]`)));`.  If we go with "raw brackets" as a new instance member in `Brackets`, what might we call that?